astroid/brain/brain_builtin_inference.py

```diff
@@ -399,6 +415,7 @@ def infer_dict(node: nodes.Call, context: InferenceContext | None = None) -> nod
     args = call.positional_arguments
     kwargs = list(call.keyword_arguments.items())
 
+    items: list[tuple[SuccessfulInferenceResult, SuccessfulInferenceResult]]
```

Comment:
Line 378 contradicts this, although that might be too broad.

Reply:
Nice catch! We should probably broaden it out. That means this is probably also too narrow:

https://github.com/pylint-dev/astroid/blob/d4f4452fe089f600bf9144ffdcd8e698816df3c2/astroid/nodes/node_classes.py#L294-L295